RecoParticleFlow/PFClusterProducer/plugins/alpaka/PFClusterSoAProducerKernel.dev.cc

```diff
@@ -0,0 +1,1652 @@
+                        pfClusteringEdgeVars.view());
+    alpaka::exec<Acc1D>(queue,
+                        make_workdiv<Acc1D>(blocks, threadsPerBlock),
+                        ECLCCCompute2{},
+                        pfRecHits.view(),
+                        pfClusteringVars.view(),
+                        pfClusteringEdgeVars.view());
+    alpaka::exec<Acc1D>(queue,
+                        make_workdiv<Acc1D>(blocks, threadsPerBlock),
+                        ECLCCCompute3{},
+                        pfRecHits.view(),
+                        pfClusteringVars.view(),
+                        pfClusteringEdgeVars.view());
+    alpaka::exec<Acc1D>(queue,
+                        make_workdiv<Acc1D>(blocks, threadsPerBlock),
+                        ECLCCFlatten{},
+                        pfRecHits.view(),
+                        pfClusteringVars.view(),
+                        pfClusteringEdgeVars.view());
+    // topoClusterContraction
+    alpaka::exec<Acc1D>(queue,
+                        make_workdiv<Acc1D>(1, threadsPerBlockForClustering),
+                        TopoClusterContraction{},
+                        pfRecHits.view(),
+                        pfClusteringVars.view(),
+                        pfClusters.view(),
+                        nSeeds.data());
+
+    // Run fillRhfIndex on serial or parallel
+    if (std::is_same_v<Device, alpaka::DevCpu>) {
+      alpaka::exec<Acc1D>(queue,
+                          make_workdiv<Acc1D>(1, 32),
+                          FillRhfIndexSerial{},
+                          pfRecHits.view(),
+                          pfClusteringVars.view(),
+                          pfrhFractions.view());
+      alpaka::exec<Acc1D>(queue,
+                          make_workdiv<Acc1D>(nRH, threadsPerBlockForClustering),
+                          FastClusterSerial{},
+                          pfRecHits.view(),
+                          params.view(),
+                          pfClusteringVars.view(),
+                          pfClusters.view(),
+                          pfrhFractions.view());
+    } else {
+      alpaka::exec<Acc2D>(queue,
+                          make_workdiv<Acc2D>({(nRH + 31) / 32, (nRH + 31) / 32}, {32, 32}),
+                          FillRhfIndex{},
+                          pfRecHits.view(),
+                          pfClusteringVars.view(),
+                          pfrhFractions.view());
+
+      alpaka::exec<Acc1D>(queue,
+                          make_workdiv<Acc1D>(nRH, threadsPerBlockForClustering),
+                          FastCluster{},
+                          pfRecHits.view(),
+                          params.view(),
+                          pfClusteringVars.view(),
+                          pfClusters.view(),
+                          pfrhFractions.view());
+      alpaka::exec<Acc1D>(
+          queue,
+          make_workdiv<Acc1D>(4, threadsPerBlockForClustering),  // uses 4 blocks to minimize memory usage
```

Comment:
Is the `4` here the same as the `4` in `make_device_buffer<...>(queue, 4 * 2304)` in the `PFClusterProducerKernel` constructor ?

If they should be the same, can you define a named constants and use it everywhere ?